src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3293,10 +3216,15 @@ def aks_pod_identity_add(cmd, client, resource_group_name, cluster_name,
         pod_identity.binding_selector = binding_selector
     pod_identities.append(pod_identity)
 
+    from azext_aks_preview.decorator import AKSPreviewModels
+
+    # store all the models used by pod identity
+    pod_identity_models = AKSPreviewModels(cmd, CUSTOM_MGMT_AKS_PREVIEW).pod_identity_models
     _update_addon_pod_identity(
         instance, enable=True,
         pod_identities=pod_identities,
         pod_identity_exceptions=instance.pod_identity_profile.user_assigned_identity_exceptions,
+        models=pod_identity_models
```

Comment:
question: since we are repeating the `pod_identity_models` construction here, why not just let the `_update_addon_pod_identity` to construct it?

Reply:
`pod_identity_models` is a dictionary stores the definition of the pod identity related models. Before that, the model definition was passed to the function through global variables (e.g. from xxxx_xx_xx import xxxx).